core/node/prover_api/src/processor.rs

```diff
@@ -0,0 +1,196 @@
+use std::sync::Arc;
+
+use axum::{extract::Path, Json};
+use zksync_basic_types::{
+    basic_fri_types::Eip4844Blobs, commitment::L1BatchCommitmentMode, L1BatchNumber,
+};
+use zksync_dal::{ConnectionPool, Core, CoreDal};
+use zksync_object_store::{bincode, ObjectStore};
+use zksync_prover_interface::{
+    api::{
+        ProofGenerationData, ProofGenerationDataRequest, ProofGenerationDataResponse,
+        VerifyProofRequest,
+    },
+    inputs::{
+        L1BatchMetadataHashes, VMRunWitnessInputData, WitnessInputData, WitnessInputMerklePaths,
+    },
+    outputs::L1BatchProofForL1,
+};
+
+use crate::error::ProcessorError;
+
+#[derive(Clone)]
+pub(crate) struct Processor {
+    blob_store: Arc<dyn ObjectStore>,
+    pool: ConnectionPool<Core>,
+    commitment_mode: L1BatchCommitmentMode,
+    last_available_batch_number: L1BatchNumber,
+    last_available_batch_data: Option<Box<ProofGenerationData>>,
+}
+
+impl Processor {
+    pub(crate) fn new(
+        blob_store: Arc<dyn ObjectStore>,
+        pool: ConnectionPool<Core>,
+        commitment_mode: L1BatchCommitmentMode,
+        last_available_batch_number: L1BatchNumber,
+    ) -> Self {
+        Self {
+            blob_store,
+            pool,
+            commitment_mode,
+            last_available_batch_number,
+            last_available_batch_data: None,
+        }
+    }
+
+    #[tracing::instrument(skip_all)]
+    pub(crate) async fn get_proof_generation_data(
+        &mut self,
+        request: Json<ProofGenerationDataRequest>,
+    ) -> Result<Json<ProofGenerationDataResponse>, ProcessorError> {
```

Comment:
My understanding was that we can give any batch that already has data required for proving (with some caveats, e.g. we probably should only expose batches for the last protocol version, for the sake of simplicity).

I would imagine the request to accept `Option<L1BatchNumber>`, where we either return data for a particular batch, or latest available.

Reply:
Huh, I thought we kinda agreed to return the latest one... At least this is what was described in the design doc. But not hard to implement